assets/js/stripe-payment-request.js

```diff
@@ -490,6 +424,148 @@ jQuery( function( $ ) {
 			} );
 		},
 
+		/**
+		 * Creates stripe paymentRequest element or connects to custom button
+		 *
+		 * @param {object} elements       Stripe elements instance.
+		 * @param {object} paymentRequest Stripe paymentRequest object.
+		 *
+		 * @return {object} Stripe paymentRequest element or custom button jQuery element.
+		 */
+		createPaymentRequestButton: function( elements, paymentRequest ) {
+			var button;
+			if ( wc_stripe_payment_request_params.button.is_custom ) {
+				button = $( wc_stripe_payment_request_params.button.css_selector );
+				if ( button.length ) {
+					// We fallback to default paymentRequest button if no custom button is found in the UI.
+					// Add flag to be sure that created button is custom button rather than fallback element.
+					button.isCustom = true;
+					return button;
+				}
+			}
+
+			return elements.create( 'paymentRequestButton', {
+				paymentRequest: paymentRequest,
+				style: {
+					paymentRequestButton: {
+						type: wc_stripe_payment_request_params.button.type,
+						theme: wc_stripe_payment_request_params.button.theme,
+						height: wc_stripe_payment_request_params.button.height + 'px',
+					},
+				},
+			} );
+		},
+
+		attachPaymentRequestButtonEventListeners: function ( prButton, paymentRequest ) {
+			if ( wc_stripe_payment_request_params.is_product_page ) {
+				wc_stripe_payment_request.attachProductPageEventListeners( prButton, paymentRequest );
+			} else {
+				wc_stripe_payment_request.attachCartPageEventListeners( prButton, paymentRequest );
+			}
+		},
+
+		attachProductPageEventListeners( prButton, paymentRequest ) {
```

Comment:
I haven't reviewed this entirely yet, but I tried to build the files from this PR and got the following error:
```
Parse error at assets/js/stripe-payment-request.js:467,33
                attachProductPageEventListeners( prButton, paymentRequest ) {
                                               ^
ERROR: Unexpected token: punc «(»
```

Should this be:
```suggestion
		attachProductPageEventListeners: function ( prButton, paymentRequest ) {
```

Reply:
Good catch! Thank you @luizreis ! Fixed in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1197/commits/f72fcd98cf859beba99c7bd182cbda24ca2f31f0. 
IMO we should migrate that code to es6 and add webpack to the project some time.